src/system-health/health_checker/service_checker.py

```diff
@@ -24,23 +43,171 @@ class ServiceChecker(HealthChecker):
 
     def __init__(self):
         HealthChecker.__init__(self)
+        self.critical_process_dict = {}
+        # Containers that has invalid critical_processes file
+        self.bad_containers = set()
+
+        self.container_feature_dict = {}
+
+        self.need_save_cache = False
+
+        self.load_critical_process_cache()
+
+    def get_expected_runnning_container_set(self, feature_table):
+        """Get a set of containers that are expected to running on SONiC
+
+        Args:
+            feature_table (object): FEATURE table in CONFIG_DB
+
+        Returns:
+            set: A set of container names
+        """
+        containers = set()
+        container_feature_dict = {}
+        for feature_name, feature_entry in feature_table.items():
+            if feature_entry["state"] not in ["disabled", "always_disabled"]:
+                if multi_asic.is_multi_asic():
+                    if feature_entry["has_global_scope"] == "True":
+                        containers.add(feature_name)
+                        container_feature_dict[feature_name] = feature_name
+                    if feature_entry["has_per_asic_scope"] == "True":
+                        num_asics = multi_asic.get_num_asics()
+                        for asic_id in range(num_asics):
+                            containers.add(feature_name + str(asic_id))
+                            container_feature_dict[feature_name + str(asic_id)] = feature_name
+                else:
+                    containers.add(feature_name)
+                    container_feature_dict[feature_name] = feature_name
+
+        return containers, container_feature_dict
+
+    def get_current_running_container_set(self):
```

Comment:
Would it be better to rename this function name to `get_current_running_containers(self)`?